operator/apis/operator/v1alpha1/satellite_types.go

```diff
@@ -0,0 +1,84 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SatelliteSpec defines the desired state of Satellite
+type SatelliteSpec struct {
+	// Version of Satellite.
+	// +kubebuilder:validation:Required
+	Version string `json:"version"`
+	// Image is the Satellite Docker image to deploy.
+	Image string `json:"image,omitempty"`
+	// Count is the number of Satellite servers
+	// +kubebuilder:validation:Required
+	Instances int32 `json:"instances"`
+	// Config holds the Satellite server configuration.
+	Config []corev1.EnvVar `json:"config,omitempty"`
+	// Service relevant settings
+	// +kubebuilder:validation:Optional
+	Service Service `json:"service,omitempty"`
```

Comment:
Could you add a `field` to set the name of an `OAPServer` which is controlled by the same operator with `Satellite`? It would simplify the onboarding process.